aws/data_source_aws_sfn_activity_test.go

```diff
@@ -0,0 +1,58 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestAccAWSStepFunctionsActivityDataSource(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAWSStepFunctionsActivityDataSourceConfig_ActivityArn(rName),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("data.aws_sfn_activity.test", "name", rName),
+					resource.TestCheckResourceAttrSet("data.aws_sfn_activity.test", "arn"),
+					resource.TestCheckResourceAttrSet("data.aws_sfn_activity.test", "creation_date"),
```

Comment:
For data source testing, its preferable to compare the state values between the resource and data source:

```go
					resource.TestCheckResourceAttrPair("data.aws_sfn_activity.test", "arn", "aws_sfn_activity.test", "arn"),
					resource.TestCheckResourceAttrPair("data.aws_sfn_activity.test", "creation_date", "aws_sfn_activity.test", "creation_date"),
					resource.TestCheckResourceAttrPair("data.aws_sfn_activity.test", "name", "aws_sfn_activity.test", "name"),

```